From ef124ad0ff8abfbf4ebe3fe6d7dcef4541dec13a Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Thu, 16 Jun 2022 18:39:21 +0200
Subject: [PATCH] ARM: dts: qcom: add krait-cache compatible for ipq806x dtsi

Add qcom,krait-cache compatible to enable cache devfreq driver for
ipq806x SoC and move the L2 node to the soc node to make the devfreq
driver correctly probe.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 arch/arm/boot/dts/qcom-ipq8064.dtsi | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

--- a/arch/arm/boot/dts/qcom-ipq8064.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8064.dtsi
@@ -69,16 +69,6 @@
 				min-residency-us = <3000>;
 			};
 		};
-
-		L2: l2-cache {
-			compatible = "cache";
-			cache-level = <2>;
-			qcom,saw = <&saw_l2>;
-
-			clocks = <&kraitcc 4>;
-			clock-names = "l2";
-			operating-points-v2 = <&opp_table_l2>;
-		};
 	};
 
 	opp_table_l2: opp_table_l2 {
@@ -1409,6 +1399,16 @@
 			#reset-cells = <1>;
 		};
 
+		L2: l2-cache {
+			compatible = "cache", "qcom,krait-cache";
+			cache-level = <2>;
+			qcom,saw = <&saw_l2>;
+
+			clocks = <&kraitcc 4>;
+			clock-names = "l2";
+			operating-points-v2 = <&opp_table_l2>;
+		};
+
 		lpass@28100000 {
 			compatible = "qcom,lpass-cpu";
 			status = "disabled";
